crates/rnote-cli/src/export.rs

```diff
@@ -0,0 +1,715 @@
+    /// Currently `.svg`, `.xopp` and `.pdf` are supported.{n}
+    /// Usages: {n}
+    /// rnote-cli export doc --output-file [filename.(svg|xopp|pdf)] [1 file]{n}
+    /// rnote-cli export doc --output-format [svg|xopp|pdf] [list of files]
+    Doc {
+        #[command(flatten)]
+        file: FileArgs<DocOutputFormat>,
+        /// The page order when documents with layouts that expand in horizontal and vertical directions are cut into pages.
+        #[arg(short = 'P', long, default_value_t = PageOrder::default())]
+        page_order: PageOrder,
+    },
+    /// Export each page of the documents individually. {n}
+    /// Both --output-dir and --output-format need to be set {n}
+    /// Usage: {n}
+    /// rnote-cli export doc-pages --output-dir [folder] --output-format [svg|png|jpeg] [list of files]
+    DocPages {
+        /// the folder the pages get exported to
+        #[arg(short = 'o', long)]
+        output_dir: PathBuf,
+        /// the folder the pages get exported to
+        #[arg(short = 's', long)]
+        output_file_stem: Option<String>,
+        /// the export output format
+        #[arg(short = 'f', long)]
+        output_format: DocPagesOutputFormat,
+        /// The page order when documents with layouts that expand in horizontal and vertical directions are cut into pages.
+        #[arg(short = 'P', long, default_value_t = PageOrder::default())]
+        page_order: PageOrder,
+        /// bitmap scale factor in relation to the actual size on the document
+        #[arg(long, default_value_t = 1.8)]
+        bitmap_scalefactor: f64,
+        /// quality of the jpeg image
+        #[arg(long, default_value_t = 85)]
+        jpeg_quality: u8,
+    },
+    /// Export selection of a document {n}
+    /// When using --output-file, only one input file can be given.{n}
+    /// The export format is recognized from the file extension of the output file.{n}
+    /// When using --output-format, the same file name is used with the extension changed.{n}
+    /// --output-file and --output-format are mutually exclusive but one of them is required.{n}
+    /// Usages: {n}
+    /// rnote-cli export doc --output-file [filename.(svg|xopp|pdf)] [1 file] selection {n}
+    /// rnote-cli export doc --output-format [svg|xopp|pdf] [list of files] selection {n}
+    /// Available selection args: only use one of them {n}
+    /// --all: select all strokes {n}
+    /// --rect X Y deltaX deltaY : Select all strokes in given area {n}
+    Selection {
+        #[command(flatten)]
+        file: FileArgs<SelectionOutputFormat>,
+        #[command(flatten)]
+        selection: SelectionArgs,
+        #[arg(short = 'i', long, default_value_t = Bounds::default(), conflicts_with = "all")]
+        /// if the lines inside or intersecting with the given bounds are exported. Exclusive with --all
+        bounds: Bounds,
+        /// bitmap scale factor in relation to the actual size on the document
+        #[arg(long, default_value_t = 1.8)]
```

Comment:
here as well: is it possible to use the default impl from SelectionExportOptions?